server/src/main/java/org/opensearch/indices/replication/RemoteSegmentFileChunkWriter.java

```diff
@@ -78,7 +78,12 @@ public void writeFileChunk(
         // Pause using the rate limiter, if desired, to throttle the recovery
         final long throttleTimeInNanos;
         // always fetch the ratelimiter - it might be updated in real-time on the recovery settings
-        final RateLimiter rl = recoverySettings.rateLimiter();
+        final RateLimiter rl;
+        if (SegmentReplicationTargetService.Actions.FILE_CHUNK.equals(action)) {
+            rl = recoverySettings.replicaitonRateLimiter();
+        } else {
+            rl = recoverySettings.recoveryRateLimiter();
+        }
```

Comment:
Can we move this to the ctor instead?

Reply:
As comment above, always fetch the newest setting to ratelimiter

Reply:
+1. This also allows us to further decouple replication & recovery components.  We can do this with a supplier:

```java
    public RemoteSegmentFileChunkWriter(
        long replicationId,
        Supplier<RateLimiter> rateLimiterProvider,
        TransportRequestOptions fileChunkRequestOptions,
        RetryableTransportClient retryableTransportClient,
        ShardId shardId,
        String action,
        AtomicLong requestSeqNoGenerator,
        Consumer<Long> onSourceThrottle
    ) {
        this.replicationId = replicationId;
        this.rateLimiterProvider = rateLimiterProvider;
        this.retryableTransportClient = retryableTransportClient;
        this.shardId = shardId;
        this.requestSeqNoGenerator = requestSeqNoGenerator;
        this.onSourceThrottle = onSourceThrottle;
        this.fileChunkRequestOptions = fileChunkRequestOptions;
        this.action = action;
    }
```

This then becomes rateLimiterProvider.get() allowing dynamic update.

 `RemoteRecoveryTargetHandler` and `SegmentReplicationSourceService` I think are the only places this is used.